redis_session.c

```diff
@@ -669,6 +736,21 @@ PS_WRITE_FUNC(redis)
     /* send SET command */
 #if (PHP_MAJOR_VERSION < 7)
     session = redis_session_key(rpm, key, strlen(key), &session_len);
+
+    /* We need to check for PHP5 if the session key changes (a bug with session_regenerate_id() is causing a missing PS_CREATE_SID call)*/
+    char *session_key_nt = estrndup(session, session_len);
+
+    int session_key_changed = strcmp(pool->lock_status->session_key, session_key_nt) != 0;
+    if (session_key_changed) {
+        efree(pool->lock_status->session_key);
+        pool->lock_status->session_key = session_key_nt;
```

Comment:
May be better to do `estrndup(session, session_len)` here?

Reply:
Sorry I do not fully get it :)

There is already a `estrndup(session, session_len);` in line 741.
Its required that `strcmp` is working correctly, as `redis_session_key()` is not null-byte terminating.

Could you please explain in more details?

Reply:
I mean you don't need to duplicate session key if it isn't changed.
```php
int session_key_changed = strncmp(pool->lock_status->session_key, session, session_len);
if (session_key_changed) {
    efree(pool->lock_status->session_key);
    pool->lock_status->session_key = estrndup(session, session_len);
}
```
This also prevent memory leak (you forgot to free allocated `session_key_nt` when `!session_key_changed`

Reply:
I see, you are right. Will be changed.